### PR TITLE
Add a polymorphic qualifier to the Value Checker

### PR DESCRIPTION
**Pull request by kelloggm**
Adds a `@PolyValue` annotation to the Value Checker and associated tests to fix typetools#1236.

Also changes commented out `@PolyMinLen` annotations in the Index JDK to `@PolyValue` and uncomments them.

### CODE REVIEW

File: framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -133,6 +138,8 @@ public ValueAnnotatedTypeFactory(BaseTypeChecker checker) {
         // this line just registers the alias. The BottomVal is never used.
         addAliasedAnnotation(MinLen.class, BOTTOMVAL);
 
+        addAliasedAnnotation(PolyAll.class, POLY);
```
Comment (smillst) [inline framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java]: PolyAll should be treated as PolyValue by the framework.  So this shouldn't be needed.

File: checker/jdk/index/src/java/lang/String.java [hunk 2]
```diff
@@ -163,7 +163,7 @@ public String() {
      * @param  original
      *         A {@code String}
      */
-    public /*!PolyMinLen*/ @PolySameLen String(/*!PolyMinLen*/ @PolySameLen String original) {
+    public @PolyValue @PolySameLen String(@PolyValue @PolySameLen String original) {
```
Comment (smillst) [inline checker/jdk/index/src/java/lang/String.java]: Technically these are Value Checker annotations, so they don't belong in the index annotated jdk. They should go in org/checkerframework/common/value/statically-executable.astub.  (Could you also move the MinLen annotations there, too?)

File: framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java (hunk 1 above)
Comment (kelloggm) [inline framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java]: This is a bug, then. I removed this and the tests that rely on them being equivalent started failing (`Polymorphic2.java` is the failing test).

Comment (smillst) [inline framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java]: Ok, I'll take a look.

**Comment by kelloggm** [comment]
@smillst I think this is ready for you to look at again.

**Comment by smillst** [comment]
I moved some of the changes to another pull request and then merged master into this branch.  Once the test pass, this pull request is ready to merge.

**Pull request merged**